routes/tasks.js

```diff
@@ -84,8 +82,8 @@ router.post('/', async (req, res) => {
   return res.status(200).json(await get_status(task_id));
 });
 
-// define a snapshot cache
-const snapshot = {};
+// declare a snapshot cache
+let snapshot;
```

Comment:
It looks like you are storing the snapshot in memory for a release where the snapshot here is an object containing a dump of each study.

This means that two simultaneous snapshots runs will collide with each other here, since it's shared memory. Perhaps you'd be better off storing each release inside its own object in the snapshot object?
```
let snapshot = {
    RE_00000001: { ... },
    RE_00000002: { ... },
}
```

And then when you finish uploading the snapshot to S3, you can do
```
snapshot[releaseId] = {}
```

That way, you won't interfere with any other running tasks.

Reply:
I initially imagined that there will be barely cases where multiple releases race against each other since the release coordinator processes one release per time. As you pointed out, however, there could be some cases where multiple people kicks in releases at the same time. I will revert the declaration of the snapshot object (`let snapshot;`) back to `const snapshot = {}`; and every action of `initialize` will create a nested object with its release ID as a key (snapshot[release_id] = {};). When cancelled, failed, or staged, only the nested object with the release ID will be flushed out (`delete snapshot[release_id];`).